tests/codebuild/feature_tests.sh

```diff
@@ -11,13 +11,50 @@ function run_feature_canary_tests
 }
 
 # Applies each of the resources needed for the integration tests.
+# Parameter:
+#    $1: CRD namespace
 function run_feature_integration_tests
 {
   echo "Running feature integration tests"
+  local crd_namespace="$1"
   run_feature_canary_tests
 
-  run_test testfiles/failing-xgboost-mnist-hpo.yaml
-  run_test testfiles/failing-xgboost-mnist-trainingjob.yaml
+  run_test "${crd_namespace}" testfiles/failing-xgboost-mnist-hpo.yaml
+  run_test "${crd_namespace}" testfiles/failing-xgboost-mnist-trainingjob.yaml
+}
+
+# Creates a training job in the specified namespace
+# For this test, the job is created in a namespace that does not have the operator. 
+# Parameter:
+#    $1: CRD namespace
+function run_feature_namespaced_tests
+{
+  echo "Running feature namespaced test"
+  local crd_namespace="$1"
+  # Create the noop-namespace if it does not exist
+  kubectl create namespace "${crd_namespace}"
+  run_feature_canary_tests
+  run_test "${crd_namespace}" testfiles/xgboost-mnist-trainingjob-namespaced.yaml  
+}
+
+# Verifies that the job created in an incorrect namespace does not gain a status or sagemaker name.
+# Parameter:
+#    $1: CRD namespace
+function verify_feature_namespaced_tests
```

Comment:
nit: Function name doesn't adequately describe what the function does.

Reply:
Fixed, but open to a better suggestion.